C/plugins/common/include/simple_http.h

```diff
@@ -0,0 +1,46 @@
+#ifndef _SIMPLE_HTTP_H
+#define _SIMPLE_HTTP_H
+/*
+ * FogLAMP HTTP Sender wrapper.
+ *
+ * Copyright (c) 2018 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Massimiliano Pinto, Mark Riddoch
+ */
+
+#include <string>
+#include <vector>
+#include <http_sender.h>
+#include <client_http.hpp>
+
+using HttpClient = SimpleWeb::Client<SimpleWeb::HTTP>;
+
+class SimpleHttp: public HttpSender
+{
+	public:
+		/**
+		 * Constructor: pass host:port & connect & request timeouts
+		 */
+		SimpleHttp(const std::string& host_port,
+				unsigned int connect_timeout = 0,
+				unsigned int request_timeout = 0);
+
+		// Destructor
+		~SimpleHttp();
+
+		/**
+		 * HTTP(S) request: pass method and path, HTTP headers and POST/PUT payload.
+		 */
+		int sendRequest(const std::string& method = std::string(HTTP_SENDER_DEFAULT_METHOD),
+				const std::string& path = std::string(HTTP_SENDER_DEFAULT_PATH),
+				const std::vector<std::pair<std::string, std::string>>& headers = {},
+				const std::string& payload = std::string());
+	private:
```

Comment:
Suggest adding private copy constructor and operator= to stop inadvertently creating copies as this will cause memory issues with m_sender

Reply:
Done